weejson/src/main/scala/com/rallyhealth/weejson/v1/BufferedValue.scala

```diff
@@ -172,15 +186,57 @@ object BufferedValue extends Transformer[BufferedValue] {
   sealed trait AnyNum extends BufferedValue {
     def value: BigDecimal
   }
+
   case class Num(s: String, decIndex: Int, expIndex: Int) extends AnyNum {
-    override def value: BigDecimal = BigDecimal(s)
+    override lazy val value: BigDecimal = BigDecimal(s)
+
+    override def equals(that: Any): Boolean = that match {
+      case NumLong(l) => value == l.value
+      case NumDouble(d) =>
+        val thisD = value.toDouble // may chop precision or go infinite
+        if (thisD.isInfinite) value == d.value else thisD == d
```

Comment:
What's the case for `Infinity` equality? `BigDecimal` doesn't seem to agree. I'm not sure we should accept loss of precision here. `BufferedValue` is supposed to be as faithful to the raw input as possible.

```
scala> val bd = BigDecimal("1e500")
val bd: BigDecimal = 1E+500

scala> bd.toDouble
val res20: Double = Infinity

scala> bd == res20
val res21: Boolean = false
```

Reply:
I'll remove it. The property tests only ensure things that should be equal are equal, but I was concerned about things being considered equal that should not be. However, there is a flaw in my logic -- `BigDecimal` can't represent an infinity anyway, so when `NumDouble` declares `override def value: BigDecimal = BigDecimal(d)`, that ain't gonna work for infinities since `BigDecimal(Double.PositiveInfinity)` throws java.lang.NumberFormatException! So, although `BigDecimal` is "larger" than `Double` in almost every sense, in this sense it is "smaller". I was jumping through invisible hoops.